spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchScan.java

```diff
@@ -130,83 +122,6 @@ public StructType readSchema() {
     return readSchema;
   }
 
-  @Override
-  public InputPartition[] planInputPartitions() {
-    String expectedSchemaString = SchemaParser.toJson(expectedSchema);
-
-    // broadcast the table metadata as input partitions will be sent to executors
-    Broadcast<Table> tableBroadcast = sparkContext.broadcast(SerializableTable.copyOf(table));
-
-    List<CombinedScanTask> scanTasks = tasks();
-    InputPartition[] readTasks = new InputPartition[scanTasks.size()];
-
-    Tasks.range(readTasks.length)
-        .stopOnFailure()
-        .executeWith(localityPreferred ? ThreadPools.getWorkerPool() : null)
-        .run(index -> readTasks[index] = new ReadTask(
-            scanTasks.get(index), tableBroadcast, expectedSchemaString,
-            caseSensitive, localityPreferred));
-
-    return readTasks;
-  }
-
-  @Override
-  public PartitionReaderFactory createReaderFactory() {
```

Comment:
I think this method got bigger and bigger over time and was fairly complicated now.